azure/activity_logs_monitoring/index.js

```diff
@@ -135,22 +135,31 @@ function extractResourceId(record) {
     metadata = { tags: [], source: '' };
     if (
         record.resourceId === undefined ||
-        typeof record.resourceId !== 'string' ||
-        !record.resourceId.toLowerCase().startsWith('/subscriptions/')
+        typeof record.resourceId !== 'string'
     ) {
         return metadata;
     }
-    var resourceId = record.resourceId.toLowerCase().split('/');
-    if (resourceId.length > 2) {
-        metadata.tags.push('subscription_id:' + resourceId[2]);
+    else if(record.resourceId.toLowerCase().startsWith('/subscriptions/')){
+      var resourceId = record.resourceId.toLowerCase().split('/');
+      if (resourceId.length > 2) {
+          metadata.tags.push('subscription_id:' + resourceId[2]);
+      }
+      if (resourceId.length > 4) {
+          metadata.tags.push('resource_group:' + resourceId[4]);
+      }
+      if (resourceId.length > 6 && resourceId[6]) {
+          metadata.source = resourceId[6].replace('microsoft.', 'azure.');
+      }
+      return metadata;
     }
-    if (resourceId.length > 4) {
-        metadata.tags.push('resource_group:' + resourceId[4]);
+    else if(record.resourceId.toLowerCase().startsWith('/tenants/')){
+      var resourceId = record.resourceId.toLowerCase().split('/');
+      if (resourceId.length > 2 && resourceId[4]) {
```

Comment:
Is there any chance the resourceId.length could be > 2 but less than 5?

Reply:
Unfortunately we don't have an exhaustive list of resourceIds so I am unsure. However in that case we just won't process the resourceId to extract source and assign 'azure' as the default source which is the behavior right now.

Reply:
I think line 157 should read:

`if (resourceId.length > 4 && resourceId[4]) {`

That would cover the situation where the resourceId is greater than 2, but less than 5.

Reply:
Fixed.